### ISSUE TITLE
Send Request ID, request status

### ISSUE THREAD

**Issue by dcposch** [daimo-eth/daimo#97]
## Goal

Show whether a request for payment has been paid.

## Current situation

Alice sends Bob a request link for $20.

Bob pays. Alice gets a notification saying Bob paid her $20.

However, payment is not tied to the request onchain. If either of them clicks the link again, it still says "Alice is requesting $20". Bob could accidentally pay the same request twice.

## Proposed solution

Add a random bytes32 requestID.

- Add a `RequestLinkScreen`. When Bob clicks the link, instead of the generic send screen with amount and recipient prefilled, he sees a `Alice is requesting $20` in-app, with a button to pay her.
- Payment is not a bare USDC transfer. Instead, goes thru Daimo router contract, which emits the requestID as filled.
  - Alternately, attach the requestID as extra data to the userop somehow
- Either way, indexer now knows which requests have been filled. (Note indexer does NOT have a list of pending, unfilled requests--those are offchain.)

**Comment by dcposch** [comment]
## Sending request ID with payment


### Option 1 - userop extradata?

...does it have that?

### Option 2 - separate RequestManager contract

All it has to do is emit RequestFulfilled events!

Recipient app validates that correct amount was fulfilled.

RequestManager is not aware of request amounts.

TLDR; RequestManager's only job is to tag USDC transfers.

#### Option 2.1 - infinite-approve ReqManager on account creation

ReqManager calls transferFrom()

...we already have to do this for other contracts including NoteManager & paymaster, so what's one more.

#### Option 2.2 - send as a multi call, 

Sender calls transfer() on USDC, fulfillReq() on ReqManager

..atomically in a single userop.

ReqManager is a pure bulletin board, never touches money.

...is there any existing contract that already does this?
"GenericEventEmitter" or "UserOpTagger" or something.

**Comment by dcposch** [comment]
## Tradeoffs

* Option 1. Pro: lowest cost. Userop exactly same size as a bare transfer. Con: only works when fulfilling the request from Daimo. If you fulfill the request using Connect Wallet from an EOA, nonce must be sequential.

* Option 2. Pro: most general (works with any account via Connect Wallet). Con: somewhat more complicated. Would need a way to so send a transaction + ERC-1271 signature without having to sign two separate messages / pop up Secure Enclave auth twice.

* Option 3. Pro: should work with any account. Con: nonstandard (is anyone else using the property that extra calldata is ignored?).

**Comment by dcposch** [comment]
done
